tests/storage/snapshot.go

```diff
@@ -733,6 +734,70 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
 				}
 			})
 
+			It("should report appropriate event when freeze fails", func() {
+				// Activate SELinux and reboot machine so we can force fsfreeze failure
+				const userData = "#cloud-config\n" +
+					"password: fedora\n" +
+					"chpasswd: { expire: False }\n" +
+					"runcmd:\n" +
+					"  - sudo sed -i 's/^SELINUX=.*/SELINUX=enforcing/' /etc/selinux/config\n" +
+					"  - sudo reboot"
+
+				vmi := libvmi.NewFedora(
+					libvmi.WithCloudInitNoCloudUserData(userData),
+					libvmi.WithNamespace(testsuite.GetTestNamespace(nil)))
+
+				dv := libdv.NewDataVolume(
+					libdv.WithBlankImageSource(),
+					libdv.WithPVC(libdv.PVCWithStorageClass(snapshotStorageClass), libdv.PVCWithVolumeSize(cd.BlankVolumeSize)),
+				)
+
+				vm = libvmi.NewVirtualMachine(vmi, libvmi.WithDataVolumeTemplate(dv))
+
+				// Adding snapshotable volume
+				vm.Spec.Template.Spec.Domain.Devices.Disks = append(vm.Spec.Template.Spec.Domain.Devices.Disks, v1.Disk{
+					Name: "blank",
+					DiskDevice: v1.DiskDevice{
+						Disk: &v1.DiskTarget{
+							Bus: v1.DiskBusVirtio,
+						},
+					},
+				})
+				vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+					Name: "blank",
+					VolumeSource: v1.VolumeSource{
+						DataVolume: &v1.DataVolumeSource{
+							Name: dv.Name,
+						},
+					},
+				})
+
+				vm, vmi = createAndStartVM(vm)
+				libwait.WaitForSuccessfulVMIStart(vmi,
+					libwait.WithTimeout(300),
+				)
+
+				Eventually(matcher.ThisVMI(vmi), 1*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+				Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+				// Recreating one specific SELinux error
+				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "mkdir /mount_dir\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: "mkfs.ext4 /dev/vda1\n"},
```

Comment:
Can we always use vda1? I recall it backfired before with https://github.com/kubevirt/kubevirt/pull/10451

Reply:
Good catch.